cpp/src/Ice/Instance.cpp

```diff
@@ -103,6 +104,36 @@ namespace
         }
     }
 
+    string getProgramName()
+    {
+        string programName{"IceC++"};
+#if defined(__APPLE__) && TARGET_OS_IPHONE == 0
+        vector<char> buffer(PATH_MAX);
+        int len = proc_pidpath(getpid(), buffer.data(), static_cast<uint32_t>(buffer.size()));
+        if (len > 0)
+        {
+            programName = string{buffer.data(), static_cast<size_t>(len)};
+        }
+#elif defined(__linux__)
+        // Read the command line from proc
+        string procExePath{"/proc/self/exe"};
+        vector<char> buffer(PATH_MAX);
+        ssize_t len = readlink(procExePath.c_str(), buffer.data(), buffer.size());
+        if (len > 0)
+        {
+            programName = string{buffer.data(), static_cast<size_t>(len)};
+        }
+#elif defined(_WIN32)
+        vector<char> buffer(MAX_PATH);
+        DWORD len = GetModuleFileNameA(NULL, buffer.data(), static_cast<DWORD>(buffer.size()));
```

Comment:
Is it correct to use `GetModuleFileNameA` which returns a string in the application native encoding, not necessarily UTF-8.

Reply:
Maybe we should use the unicode variant `GetModuleFileNameW`?

Reply:
You are right, parseLine does the conversion to native.